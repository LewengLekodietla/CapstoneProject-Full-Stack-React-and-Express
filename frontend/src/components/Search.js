import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import api from '../services/api.js';

const Search = () => {
  // Local state to capture the user's input
  const [username, setUsername] = useState("");

  // React Router hook for navigation
  const navigate = useNavigate();

  // Handle form submission to search for a GitHub user
  const handleSubmit = async (e) => {
    e.preventDefault(); // Prevent default form behavior (page reload)
    if (username.trim()) {
      try {
        const user = await api.fetchUserData(username); // Fetch the user using the inputted username
        console.log("User fetched:", user); // Optional debug log
        navigate(`/user/${username}`);
      } catch (error) {
        console.error(error); // Log error if user not found or fetch fails
      }
    }
  };

  return (
    <div>
      <h1>GitHub User Search</h1>
      <form onSubmit={handleSubmit}>
        <input
          value={username}
          onChange={(e) => setUsername(e.target.value)}
          placeholder="Enter Github username"
        />
        <button type="submit">Search</button>
      </form>
    </div>
  );
};

export default Search;