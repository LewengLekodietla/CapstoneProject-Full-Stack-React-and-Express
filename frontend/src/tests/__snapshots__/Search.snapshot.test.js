import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Search from '../../components/Search';

// Snapshot test for Search component to ensure consistent UI rendering
it('matches snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Search />
    </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});