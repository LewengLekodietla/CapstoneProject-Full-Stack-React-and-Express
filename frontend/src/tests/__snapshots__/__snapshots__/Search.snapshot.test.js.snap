// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    GitHub User Search
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter Github username"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
